tests/components/intellifire/test_config_flow.py

```diff
@@ -200,10 +260,56 @@ async def test_picker_already_discovered(
             CONF_HOST: "192.168.1.4",
         },
     )
-    assert result2["type"] == RESULT_TYPE_CREATE_ENTRY
-    assert result2["title"] == "Fireplace 12345"
-    assert result2["data"] == {CONF_HOST: "192.168.1.4"}
-    assert len(mock_setup_entry.mock_calls) == 2
+    assert result2["type"] == RESULT_TYPE_FORM
+    assert len(mock_setup_entry.mock_calls) == 0
+
+
+@patch.multiple(
+    "homeassistant.components.intellifire.config_flow.IntellifireControlAsync",
+    login=AsyncMock(),
+    get_username=AsyncMock(return_value="intellifire"),
+)
+async def test_reauth_flow(
+    hass: HomeAssistant,
+    mock_setup_entry: AsyncMock,
+    mock_intellifire_config_flow: MagicMock,
+) -> None:
+    """Test the reauth flow."""
+
+    entry = MockConfigEntry(
+        domain=DOMAIN,
+        data={
+            "host": "192.168.1.3",
+        },
+        title="Fireplace 1234",
+        version=1,
+        unique_id="4444",
+    )
+    entry.add_to_hass(hass)
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={
+            "source": "reauth",
+            "unique_id": entry.unique_id,
+            "entry_id": entry.entry_id,
+        },
+    )
+
+    assert result["type"] == RESULT_TYPE_FORM
+    assert result["step_id"] == "api_config"
+
+    with patch(
+        "homeassistant.config_entries.ConfigFlow.async_set_unique_id",
```

Comment:
Why do we need to patch this? We've added a mock config entry so setting the unique id should return that mock config entry if we set the same unique id.